### PR TITLE
Adds a MVP for Manual Approval Gate hooked with the API service

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
Signed-off-by: Puneet Punamiya ppunamiy@redhat.com

### CODE REVIEW

**Review by vdemeester — COMMENTED** [review]
Good start 👼🏼 
I think we need to setup some e2e tests as well as unit tests 🙃

File: .github/workflows/latest.yml [hunk 1]
```diff
@@ -52,15 +57,22 @@ jobs:
...
+    - uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
     - name: install manual-approval-gate custom task
       run: |
         # kubectl create namespace tekton-pipelines
-        ko apply --local -f config/kubernetes
+        ko apply -f config/kubernetes
```
Comment (vdemeester) [inline .github/workflows/latest.yml:73]: you would need to use ` --base-import-paths` for this, as well as, probably using the `commit` sha as a tag or something to make sure we don't override `:latest`.

File: test/e2e-test.sh [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  kubectl delete --ignore-not-found=true ${res}.tekton.dev --all
+  done
+
+  # Wait for pods to be running in the namespaces we are deploying to
+  wait_until_pods_running tekton-pipelines || fail_test "Tekton Pipeline did not come up"
+}
+
+function install_manual_approval_crd() {
+  echo ">> Deploying Manual Approval Gate"
+  ko apply -f config/kubernetes
```
Comment (vdemeester) [inline test/e2e-test.sh:26]: you need the same `ko` flags here I think.

Comment (PuneetPunamiya) [reply test/e2e-test.sh:26]: So I have not called this function in the main, meaning i have commented that, directly used in the actions to check if the pods are getting up

**Pull request merged**